Tallies/TallyMaps/Maps1D/directionMap_class.f90

```diff
@@ -0,0 +1,194 @@
+module directionMap_class
+
+  use numPrecision
+  use universalVariables, only : valueOutsideArray, X_AXIS, Y_AXIS, Z_AXIS
+  use genericProcedures,  only : fatalError, dotProduct, numToChar
+  use dictionary_class,   only : dictionary
+  use grid_class,         only : grid
+  use particle_class,     only : particleState
+  use outputFile_class,   only : outputFile
+  use tallyMap1D_inter,   only : tallyMap1D, kill_super => kill
+
+  implicit none
+  private
+
+  !!
+  !! Maps the particle direction in linear bins in the range 0-360 degrees
+  !!
+  !! Interface:
+  !!   tallyMap1D interface
+  !!
+  !! NOTE:
+  !!   Behaviour of points exactly at the boundary of bins is undefined.
+  !!   particle can end-up in either of the two
+  !!
+  !! Sample Dictionary Input:
+  !!  directionMap {
+  !!    type directionMap;
+  !!    #plane xz;#            // Optional. Default xy
+  !!    N 10;
+  !!    }
+  !!
+  type, public, extends (tallyMap1D) :: directionMap
+    private
+    type(grid)                  :: bounds
+    integer(shortInt)           :: N = 0
+    integer(shortInt)           :: DIM1 = 0
+    integer(shortInt)           :: DIM2 = 0
+
+  contains
+    ! Superclass
+    procedure :: init
+    procedure :: bins
+    procedure :: getAxisName
+    procedure :: map
+    procedure :: print
+    procedure :: kill
+
+  end type directionMap
+
+contains
+
+  !!
+  !! Initialise tallyMap from a dictionary
+  !!
+  !! See tallyMap for specification.
+  !!
+  subroutine init(self, dict)
+    class(directionMap), intent(inout)       :: self
+    class(dictionary), intent(in)            :: dict
+    character(nameLen)                       :: type
+    character(100), parameter :: Here = 'init (directionMap_class.f90)'
+
+    ! Check orientation of the cylinder
+    if (dict % isPresent('plane')) then
+      call dict % get(type, 'plane')
+    else
+      type = 'xy'
+    end if
+
+    select case(type)
+      case('yz')
+        self % DIM1 = Y_AXIS
+        self % DIM2 = Z_AXIS
+
+      case('xz')
+        self % DIM1 = X_AXIS
+        self % DIM2 = Z_AXIS
+
+      case('xy')
+        self % DIM1 = X_AXIS
+        self % DIM2 = Y_AXIS
+
+      case default
+        call fatalError(Here, 'Keyword orientation must be x, y or z. It is: '//type)
+
+    end select
+
+    ! Build grid
+    call dict % get(self % N, 'N')
+    call self % bounds % init(-PI, PI, self % N, 'lin')
```

Comment:
Why from -pi to pi rather than 0 to 2pi? I would say the latter is more common in describing the azimuthal angle.

Reply:
Just for convenience, because the Fortran function atan2 returns an angle in the range [-PI, PI].